android/KMEA/app/src/main/java/com/tavultesoft/kmea/packages/LexicalModelPackageProcessor.java

```diff
@@ -0,0 +1,228 @@
+   * @param path The file path of the .kmp file, file name included.
+   * @return The mapped temporary file path for the .kmp file's contents.
+   * @throws IOException
+   */
+  public static File unzipKMP(File path) throws IOException {
+    File tempModelPath = constructPath(path, true);
+    if (!tempModelPath.exists()) {
+      tempModelPath.mkdir();
+    }
+    ZipUtils.unzip(path, tempModelPath);
+
+    return tempModelPath;
+  }
+
+  static boolean lexicalModelExists(final String packageId, final String modelId) {
+    if (resourceRoot != null) {
+      FileFilter lexicalModelFilter = new FileFilter() {
+        @Override
+        public boolean accept(File pathname) {
+          if (pathname.isFile() && pathname.getName().startsWith(modelId) && FileUtils.hasLexicalModelExtension(pathname.getName())) {
+            return true;
+          }
+          return false;
+        }
+      };
+
+      File kmpFile = new File(packageId + ".kmp");
+      if (!kmpFile.exists()) {
+        kmpFile = new File(packageId + ".model.kmp");
+      }
+      File packageDir = constructPath(kmpFile, false);
+      File[] files = packageDir.listFiles(lexicalModelFilter);
+      if (files.length > 0) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  static boolean welcomeExists(final String packageId) {
+    if (resourceRoot != null) {
+      FileFilter welcomeFilter = new FileFilter() {
+        @Override
+        public boolean accept(File pathname) {
+          if (pathname.isFile() && FileUtils.isWelcomeFile(pathname.getName())) {
+            return true;
+          }
+          return false;
+        }
+      };
+
+      File kmpFile = new File(packageId + ".kmp");
+      if (!kmpFile.exists()) {
+        kmpFile = new File(packageId + ".model.kmp");
+      }
+      File packageDir = constructPath(kmpFile, false);
+      File[] files = packageDir.listFiles(welcomeFilter);
+      if (files.length > 0) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  public static Map<String, String>[] processLexicalModelsEntry(JSONObject jsonModel, String packageId) throws JSONException {
+    JSONArray languages = jsonModel.getJSONArray("languages");
+
+    String modelId = jsonModel.getString("id");
+    if (lexicalModelExists(packageId, modelId)) {
+      HashMap<String, String>[] models = new HashMap[languages.length()];
+
+      for (int i = 0; i < languages.length(); i++) {
+        models[i] = new HashMap<>();
+        models[i].put(KMManager.KMKey_PackageID, packageId);
+        models[i].put(KMManager.KMKey_LexicalModelName, jsonModel.getString("name"));
+        models[i].put(KMManager.KMKey_LexicalModelID, jsonModel.getString("id"));
+        models[i].put(KMManager.KMKey_LexicalModelVersion, jsonModel.getString("version"));
+        models[i].put(KMManager.KMKey_LanguageID, languages.getJSONObject(i).getString("id"));
+        models[i].put(KMManager.KMKey_LanguageName, languages.getJSONObject(i).getString("name"));
```

Comment:
Do we want any optional fields like authorName or authorEmail?

Reference: https://help.keyman.com/DEVELOPER/cloud/model_info/1.0/